cli/azd/pkg/tools/git/git.go

```diff
@@ -96,7 +96,10 @@ func (cli *gitCli) ShallowClone(ctx context.Context, repositoryPath string, bran
 	}
 	args = append(args, target)
 
-	runArgs := newRunArgs(args...)
+	// Do not call `newRunArgs()` here because we don't want to apply the codespaces special patch that removes
+	// default authentication. `git clone` should work for private repos within a codespace with default auth.
+	// See: https://github.com/Azure/azure-dev/issues/2582
+	runArgs := exec.NewRunArgs("git", args...)
```

Comment:
YMMV: but this seems likely to break again when we add new commands. The token credential thing seems pervasive, naming or not making implicit may be a good short-term improvement. Long-term, I really wonder if we want to uphold the design here.

Were we trying to add token suppression to all commands because of cross-repo modification? From the original PR [comment](https://github.com/Azure/azure-dev/pull/1970#discussion_r1170378048):
> The other common issue was the common inability to push changes to the just created repo... because git is using the authentication provided by codespaces which is limited to read/write only for the current repo.

This just isn't supported by `github` by default, isn't it? If you run `gh` in a given repo trying to create a secondary repo, I can't imagine that's expected to succeed.

Reply:
gh won't let you create/modify a secondary repo.  However, you can `clone` as many repos as you want inside your codespace, either using `git` or `gh`.  The read access should work for public repos and any private repo within the user's account